crates/comm/src/packet.rs

```diff
@@ -66,16 +69,26 @@ impl From<u8> for NetworkPacketType {
 
 impl NetworkPacket {
```

Comment:
do you want a struct to be returned with the decompressed data or just the decompressed data itself?

Reply:
I think just a `NetworkPacket::decompress() -> Vec<u8>` since we'll need this in the `Peer` for sending packets on. Might be worth checking if we're attempting to decode with the compressed data atm.